packages/react-reconciler/src/__tests__/ReactLazy-test.internal.js

```diff
@@ -1089,4 +1089,85 @@ describe('ReactLazy', () => {
       expect(Scheduler).toFlushAndYield([]);
     }).toWarnDev('Function components cannot be given refs');
   });
+
+  it('should error with a component stack naming the resolved component', async () => {
+    let componentStackMessage;
+
+    const LazyText = lazy(() =>
+      fakeImport(function ResolvedText() {
+        throw new Error('oh no');
+      }),
+    );
+
+    class ErrorBoundary extends React.Component {
+      state = {error: null};
+
+      componentDidCatch(error, errMessage) {
+        componentStackMessage = errMessage.componentStack;
+        this.setState({
+          error,
+        });
+      }
+
+      render() {
+        return this.state.error ? null : this.props.children;
+      }
+    }
+
+    ReactTestRenderer.create(
+      <ErrorBoundary>
+        <Suspense fallback={<Text text="Loading..." />}>
+          <LazyText text="Hi" />
+        </Suspense>
+      </ErrorBoundary>,
+      {unstable_isConcurrent: true},
+    );
+
+    expect(Scheduler).toFlushAndYield(['Loading...']);
+
+    try {
+      await Promise.resolve();
+    } catch (e) {}
+
+    expect(Scheduler).toFlushAndYield([]);
+
+    expect(componentStackMessage).toContain('in ResolvedText');
+  });
+
+  it('should error with a component stack containing Lazy if unresolved', () => {
+    let componentStackMessage;
+
+    const LazyText = lazy(() => ({
+      then(resolve, reject) {
+        reject(new Error('oh no'));
+      },
+    }));
+
+    class ErrorBoundary extends React.Component {
+      state = {error: null};
+
+      componentDidCatch(error, errMessage) {
+        componentStackMessage = errMessage.componentStack;
+        this.setState({
+          error,
+        });
+      }
+
+      render() {
+        return this.state.error ? null : this.props.children;
+      }
+    }
+
+    ReactTestRenderer.create(
+      <ErrorBoundary>
+        <Suspense fallback={<Text text="Loading..." />}>
+          <LazyText text="Hi" />
+        </Suspense>
+      </ErrorBoundary>,
+    );
+
+    expect(Scheduler).toHaveYielded([]);
+
+    expect(componentStackMessage).toContain('in Lazy');
+  });
```

Comment:
You'd probably want to add an additional test case for the named lazy:
```js
it('should error with a component stack containing Lazy with displayName if unresolved', () => {
  let componentStackMessage;

  const LazyText = lazy(() => ({
    then(resolve, reject) {
      reject(new Error('oh no'));
    },
  }));
  LazyText.displayName = 'LazyText';

  class ErrorBoundary extends React.Component {
    state = {error: null};

    componentDidCatch(error, errMessage) {
      componentStackMessage = errMessage.componentStack;
      this.setState({
        error,
      });
    }

    render() {
      return this.state.error ? null : this.props.children;
    }
  }

  ReactTestRenderer.create(
    <ErrorBoundary>
      <Suspense fallback={<Text text="Loading..." />}>
        <LazyText text="Hi" />
      </Suspense>
    </ErrorBoundary>,
  );

  expect(Scheduler).toHaveYielded([]);

  expect(componentStackMessage).toContain('in LazyText');
});
```

Reply:
@motiz88 Wanna do this?

Reply:
By "this" do you mean adding back `displayName` support? Yes, I'd be happy to.

Reply:
Hmm I misread the test. I thought @bvaughn meant adding a test that verifies a *resolved* `Lazy` shows the inner component name in the stack.